Unspecific type hints for reader.metadata

Take the below example file:

```python
from PyPDF2 import PdfReader

with open("example.pdf", "rb") as fp:
    reader = PdfReader(fp)
    metadata = reader.metadata
    assert metadata is not None
    date_str = metadata["/CreationDate"]
    date_str = date_str.removeprefix("D:").replace("'", "")
    print(date_str)
```

It runs fine:

```
$ python example.py
20220415093243+0200
```

but Mypy complains about using `remove_prefix()` on `date_str`:

```
$ mypy example.py
example.py:8: error: "PdfObject" has no attribute "removeprefix"  [attr-defined]
Found 1 error in 1 file (checked 1 source file)
```

This is due to `DocumentInformation` being a subclass of `DictionaryObject`, and thus only guaranteeing that the values returned are `PdfObject`s. In practice they seem to only be `TextStringObject`s, which subclass `str`. If they're always `TextStringObject`s, the types in `DocumentInformation` should be adjusted accordingly.

## Environment

```bash
$ python -m platform
macOS-12.5-arm64-arm-64bit

$ python -c "import PyPDF2;print(PyPDF2.__version__)"
2.9.0
```

## Code + PDF

above, used `metadata.pdf` from PyPDF2 resources

--- a/PyPDF2/_reader.py
+++ b/PyPDF2/_reader.py
@@ -31,6 +31,7 @@ import os
 import re
 import struct
 import zlib
+from datetime import datetime
 from io import BytesIO
 from pathlib import Path
 from typing import (
@@ -231,33 +232,45 @@ class DocumentInformation(DictionaryObject):
         return self.get(DI.PRODUCER)
 
     @property
-    def creation_date(self) -> Optional[str]:
+    def creation_date(self) -> Optional[datetime]:
         """
         Read-only property accessing the document's **creation date**.
-
-        Typically in the format D:YYYYMMDDhhmmss[+-]hh'mm where the suffix is the
-        offset from UTC.
         """
-        return self._get_text(DI.CREATION_DATE)
+        text = self._get_text(DI.CREATION_DATE)
+        if text is None:
+            return None
+        return datetime.strptime(text.replace("'", ""), "D:%Y%m%d%H%M%S%z")
 
     @property
     def creation_date_raw(self) -> Optional[str]:
-        """The "raw" version of creation date; can return a ``ByteStringObject``."""
-        return self.get(DI.CREATION_DATE)
-
-    @property
-    def modification_date(self) -> Optional[str]:
         """
-        Read-only property accessing the document's **modification date**.
+        The "raw" version of creation date; can return a ``ByteStringObject``.
 
         Typically in the format D:YYYYMMDDhhmmss[+-]hh'mm where the suffix is the
         offset from UTC.
         """
-        return self._get_text(DI.MOD_DATE)
+        return self.get(DI.CREATION_DATE)
+
+    @property
+    def modification_date(self) -> Optional[datetime]:
+        """
+        Read-only property accessing the document's **modification date**.
+
+        The date and time the document was most recently modified.
+        """
+        text = self._get_text(DI.MOD_DATE)
+        if text is None:
+            return None
+        return datetime.strptime(text.replace("'", ""), "D:%Y%m%d%H%M%S%z")
 
     @property
     def modification_date_raw(self) -> Optional[str]:
-        """The "raw" version of modification date; can return a ``ByteStringObject``."""
+        """
+        The "raw" version of modification date; can return a ``ByteStringObject``.
+
+        Typically in the format D:YYYYMMDDhhmmss[+-]hh'mm where the suffix is the
+        offset from UTC.
+        """
         return self.get(DI.MOD_DATE)
 
 

--- a/PyPDF2/_reader.py
+++ b/PyPDF2/_reader.py
@@ -230,6 +230,36 @@ class DocumentInformation(DictionaryObject):
         """The "raw" version of producer; can return a ``ByteStringObject``."""
         return self.get(DI.PRODUCER)
 
+    @property
+    def creation_date(self) -> Optional[str]:
+        """
+        Read-only property accessing the document's **creation date**.
+
+        Typically in the format D:YYYYMMDDhhmmss[+-]hh'mm where the suffix is the
+        offset from UTC.
+        """
+        return self._get_text(DI.CREATION_DATE)
+
+    @property
+    def creation_date_raw(self) -> Optional[str]:
+        """The "raw" version of creation date; can return a ``ByteStringObject``."""
+        return self.get(DI.CREATION_DATE)
+
+    @property
+    def modification_date(self) -> Optional[str]:
+        """
+        Read-only property accessing the document's **modification date**.
+
+        Typically in the format D:YYYYMMDDhhmmss[+-]hh'mm where the suffix is the
+        offset from UTC.
+        """
+        return self._get_text(DI.MOD_DATE)
+
+    @property
+    def modification_date_raw(self) -> Optional[str]:
+        """The "raw" version of modification date; can return a ``ByteStringObject``."""
+        return self.get(DI.MOD_DATE)
+
 
 class PdfReader:
     """

--- a/tests/test_reader.py
+++ b/tests/test_reader.py
@@ -95,6 +95,10 @@ def test_read_metadata(pdf_path, expected):
         docinfo.producer_raw
         docinfo.subject
         docinfo.subject_raw
+        docinfo.creation_date
+        docinfo.creation_date_raw
+        docinfo.modification_date
+        docinfo.modification_date_raw
         if "/Title" in metadict:
             assert metadict["/Title"] == docinfo.title
 

